karapace/karapacemetrics.py

```diff
@@ -0,0 +1,169 @@
+"""
+karapace - metrics
+Supports collection of system metrics
+list of supported metrics:
+connections-active - The number of active HTTP(S) connections to server.
+                     Data collected inside aiohttp request handler.
+
+Copyright (c) 2023 Aiven Ltd
+See LICENSE for details
+"""
+from datetime import datetime
+from kafka.metrics import MetricName, Metrics
+from kafka.metrics.measurable_stat import AbstractMeasurableStat
+from kafka.metrics.stats import Avg, Max, Rate, Total
```

Comment:
issue: I don't think this is meant to be a public API, as there is no trace of this in documentation. These should probably be seen as internal implementation details within kafka-python, and I don't think we should use them.

We are having internal discussions about replacing kafka-python, which is another reason not to tie us closer to it.

Reply:
In my previous pull request, I had implemented most of the metrics using my own approach similar to kafka java classes. However, I discovered that these classes in kafka-python were essentially ports of Kafka Java classes, which are also utilized by a competing product. Considering the importance of compatibility, I made the decision to create the metrics using the same approach.

Reply:
Thanks for explaining your reasoning, however, this in my opinion is a blocking issue. We should not add these new dependencies on private APIs.

Reply:
Additionally, since kafka-python is also released under the Apache License, the licensing aspect should not pose a blocking issue either.

Reply:
This is the API reference of kafka-python: https://kafka-python.readthedocs.io/en/master/apidoc/modules.html

Regardless of what's exposed in Java, the APIs used here are not documented in kafka-python, and hence, are to be considered private implementation details.